extide/gradle/src/org/netbeans/modules/gradle/api/GradleProjects.java

```diff
@@ -88,6 +93,36 @@ public static File getJavadoc(File binary) {
         return Collections.unmodifiableMap(ret);
     }
 
+    /**
+     * Returns the NetBeans project representation of the given Gradle project.
+     * If the given project is not a Gradle project it returns <code>null</code>,
+     * otherwise returns the Project which is assigned for the given project
+     * root directory.
+     *
+     * @param project the project to determine of its root.
+     *
+     * @return the root project or <code>null</code> if the given project is not
+     *         a Gradle project.
+     * @since 2.7
+     */
+    public static Project rootOf(Project project) {
```

Comment:
It might be more useful to return `project` if no parent/root project is known. Especially if:

This might be a useful concept in generic Project API. There already is [SubprojectProvider](https://bits.netbeans.org/12.0/javadoc/org-netbeans-modules-projectapi/org/netbeans/spi/project/SubprojectProvider.html) and [ProjectContainerProvider](https://bits.netbeans.org/12.0/javadoc/org-netbeans-modules-projectapi/org/netbeans/spi/project/ProjectContainerProvider.html). There could be `ParentProvider` and/or `RootProjectProvider` and the `rootOf` method could be moved to [ProjectUtils](https://bits.netbeans.org/12.0/javadoc/org-netbeans-modules-projectapi/org/netbeans/api/project/ProjectUtils.html). Then it could/should be implemented by Maven as well.

I can imagine a "Build All" action that would build the root project to make sure all bits are compiled and then continue with some other operation.

Reply:
That's a good idea, I could implement that as well.